crates/swc_ecma_transforms_base/src/hygiene/tests.rs

```diff
@@ -1756,3 +1756,71 @@ fn issue_2539() {
         ",
     );
 }
+
+#[test]
+fn fn_scope_collision() {
+    test(
+        |tester| {
+            let marks = vec![
+                Mark::root(),
+                Mark::fresh(Mark::root()),
+                Mark::fresh(Mark::root()),
+                Mark::fresh(Mark::root()),
+            ];
+
+            let mut stmts = tester.parse_stmts(
+                "actual1.js",
+                "
+                    const i$1 = 'abc';
+                    const env$1 = i$1;
+                    ;
+                    const i$2 = env$1;
+                    ;
+                    ;
+                    const increment$3 = (i1$3)=>i1$3 + 1;
```

Comment:
The test case was wrong.
```
                    const i$1 = 'abc';
                    const env$1 = i$1;
                    ;
                    const i$2 = env$1;
                    ;
                    ;
                    const increment$3 = (i1$3)=>i1$3 + 1;
                    it$3('should correctly handle variable collisions', ()=>{
                        let i1$3 = 123;
                        expect$3(i$2).toBe('abc');
                    });
```

There are two different variables named i1$,3 but those variables are different